internal/store/enrollmentrequest.go

```diff
@@ -121,24 +123,73 @@ func (s *EnrollmentRequestStore) CreateOrUpdate(ctx context.Context, orgId uuid.
 	enrollmentrequest.OrgID = orgId
 
 	created := false
-	findEnrollmentRequest := model.EnrollmentRequest{
-		Resource: model.Resource{OrgID: orgId, Name: *resource.Metadata.Name},
-	}
-	result := s.db.First(&findEnrollmentRequest)
-	if result.Error != nil {
-		if result.Error == gorm.ErrRecordNotFound {
+	var existingRecord *model.EnrollmentRequest
+
+	err := s.db.Transaction(func(innerTx *gorm.DB) (err error) {
```

Comment:
Actually what we are doing here is select, and then insert or update - meaning that we first find if the record exists and if it does not exist then it is created, otherwise it is updated.
Even if it is all done in transaction, if the initial select statement doesn't have the "FOR UPDATE" clause, then the updated record might not be identical to the queried record which is no desirable.
Actually we can avoid doing the whole thing in transaction if we will use upsert (https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-upsert/)  which means that the whole operation is done in a single statement.  This is also supported by gorm https://gorm.io/docs/create.html#Upsert-x2F-On-Conflict